graylog2-server/src/main/java/org/graylog2/rest/resources/streams/rules/StreamRuleResource.java

```diff
@@ -192,4 +194,19 @@ public void delete(@ApiParam(name = "streamid", value = "The stream id this new
             throw new NotFoundException();
         }
     }
+
+    @GET
+    @Path("/types")
+    @Timed
+    @ApiOperation(value = "Get all available stream types")
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<StreamRuleTypeResponse> types(@ApiParam(name = "streamid", value = "The stream id this new rule belongs to.", required = true)
```

Comment:
This method should probably check whether the user is actually allowed to access the stream.